sdl-core/src/main/scala/io/smartdatalake/util/hdfs/Partition.scala

```diff
@@ -137,6 +137,17 @@ private[smartdatalake] object PartitionValues {
     diffPartitionValues(existingPartitionValues, expectedPartitionValues, partitionColCombinations)
   }
 
+  /**
+   * Read DataFrame and convert to PartitionValues
+   * @param df DataFrame with partition columns only selected. All columns must be of type string.
```

Comment:
It's possible to give something else than string, right? If it's really a "must", then you could insert an assert to check.
```suggestion
   * @param df DataFrame with partition columns only selected. All columns will be handled as string.
```

Reply:
Right, done.